python_modules/dagster-graphql/dagster_graphql/schema/pipelines/pipeline.py

```diff
@@ -612,6 +614,97 @@ def resolve_rootConcurrencyKeys(self, graphene_info: ResolveInfo):
         return root_concurrency_keys
 
 
+class GrapheneRunType(graphene.Enum):
+    """The underlying kind of run represented by a MegaRun."""
+
+    RUN = "RUN"
+    BACKFILL = "BACKFILL"
+
+    class Meta:
+        name = "RunType"
+
+
+class GrapheneMegaRun(graphene.ObjectType):
```

Comment:
Naming thread! what do we want to call this?

- MegaRun
- BackfillOrRun
- LeadRun
- RunSet
- ???

Reply:
instead of a new concrete type, it may be better to use an interface or a union and return the existing underlying types for run and backfill

https://graphql.com/learn/interfaces-and-unions/

Reply:
Union is easiest from server but then client has to fork all rendering logic based on type , interface allows for a common set of fields to handle in one place. Talking with @bengotow and looking at what fields are common probably best way to navigate. Can always add new fields to the two concrete types to support the interface as well .

Reply:
updated to use an interface